### PR TITLE
add snapshot

### PR DESCRIPTION
**Pull request by tomdong2010**

### CODE REVIEW

File: mainnet/snapshot/backup.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+DATA_DIR=$GOPATH/src/github.com/QuarkChain/goquarkchain/cmd/cluster/qkc-data/
+DATE=`date +%Y-%m-%d.%H:%M:%S`
+BACKUP_DIR=/home/ubuntu/backup
+OUTPUT_FILE=$BACKUP_DIR/$DATE.tar.gz
+LATEST_FILE=$BACKUP_DIR/LATEST
+# 3 day's backup
+RETENTION=6
+cd $DATA_DIR
+cd ..
+./stop.sh
```
Comment (syntrust) [inline mainnet/snapshot/backup.sh]: stop.sh ready?

File: mainnet/snapshot/backup.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+mkdir -p $BACKUP_DIR
+cd $DATA_DIR
+tar cvfz $OUTPUT_FILE ./mainnet/
+cd ..
+./run.sh
```
Comment (syntrust) [inline mainnet/snapshot/backup.sh]: run.sh ready?

File: mainnet/snapshot/backup.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+# script to sync data into s3:
+
+set -ex
+
+DATA_DIR=$GOPATH/src/github.com/QuarkChain/goquarkchain/cmd/cluster/qkc-data/
+DATE=`date +%Y-%m-%d.%H:%M:%S`
+BACKUP_DIR=/home/ubuntu/backup
```
Comment (syntrust) [inline mainnet/snapshot/backup.sh]: /home/ubuntu/backup hardcoded?

File: mainnet/snapshot/backup.sh (hunk 1 above)
Comment (tomdong2010) [reply mainnet/snapshot/backup.sh]: yes

File: mainnet/snapshot/backup.sh (hunk 2 above)
Comment (tomdong2010) [reply mainnet/snapshot/backup.sh]: yes

**Comment by tomdong2010** [comment]
Add file run.sh and stop or not  ?

File: mainnet/snapshot/run.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
+kill -9 `ps -ef | grep cluster_config | awk '{print $2}'`
```
Comment (syntrust) [inline mainnet/snapshot/run.sh]: remove -9, can we?

File: mainnet/snapshot/stop.sh [hunk 5]
```diff
@@ -0,0 +1,4 @@
+kill -9 `ps -ef | grep cluster_config | awk '{print $2}'`
```
Comment (syntrust) [inline mainnet/snapshot/stop.sh]: remove -9, can we?

File: mainnet/snapshot/run.sh (hunk 4 above)
Comment (tomdong2010) [reply mainnet/snapshot/run.sh]: @syntrust  why

File: mainnet/snapshot/stop.sh (hunk 5 above)
Comment (syntrust) [reply mainnet/snapshot/stop.sh]: -9 will not close db

Comment (tomdong2010) [reply mainnet/snapshot/stop.sh]: ok

File: mainnet/snapshot/backup.sh [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+fi
+DATA_DIR=$GOPATH/src/github.com/QuarkChain/goquarkchain/cmd/cluster/qkc-data/
+DATE=`date +%Y-%m-%d.%H:%M:%S`
+OUTPUT_FILE=$BACKUP_DIR/$DATE.tar.gz
+LATEST_FILE=$BACKUP_DIR/LATEST
+# 3 day's backup
+RETENTION=3
+cd $DATA_DIR
+cd ..
+./stop.sh
```
Comment (syntrust) [inline mainnet/snapshot/backup.sh]: stop.sh is not here

File: mainnet/snapshot/backup.sh [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+mkdir -p $BACKUP_DIR
+cd $DATA_DIR
+tar cvfz $OUTPUT_FILE ./mainnet/
+cd ..
+./run.sh
```
Comment (syntrust) [inline mainnet/snapshot/backup.sh]: run.sh is not here

File: mainnet/snapshot/backup.sh (hunk 6 above)
Comment (tomdong2010) [reply mainnet/snapshot/backup.sh]: why merge

Comment (tomdong2010) [reply mainnet/snapshot/backup.sh]: @mask-pp   I tried "kill -9" and "kill"  ,it showed "kill -9" that was good effect

**Pull request merged**